src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,148 @@
+
+        case "delete":
+            return new DeleteCommand(taskList, storage, ui, Integer.parseInt(splitUserInput[1]));
+
+        case "find":
+            return new FindCommand(taskList, storage, ui, userInput.substring(userInput.indexOf(" ") + 1));
+
+        default:
+            throw new NoSuchCommandException(ui);
+        }
+    }
+
+    /**
+     *
+     * @param taskType  The type of task.
+     * @param userInput User's input on what they want duke to do.
+     * @return The respective Task.
+     * @throws DukeException If description of task is invalid.
+     */
+    public Task parseTaskInput(TaskType taskType, String userInput) throws DukeException {
+        String[] splitUserInput = userInput.split(" ");
+        String commandDetails = userInput.substring(userInput.indexOf(" ") + 1);
+        if (commandDetails.isBlank() || userInput.indexOf(" ") == -1) {
+            throw new NoTaskDescriptionException(ui);
+        }
+
+        switch (taskType) {
+        case DEADLINE:
+            try {
+                int byIndex = commandDetails.indexOf("/by") - 1 ;
+                if (byIndex <= 0) {
+                    throw new NoDateTimeException(ui);
+                }
+                String deadlineDetails = commandDetails.substring(0, byIndex);
+                if (deadlineDetails.isBlank()) {
+                    throw new NoTaskDescriptionException(ui);
+                }
+                try {
+                    LocalDateTime by = LocalDateTime.parse(commandDetails.substring(deadlineDetails.indexOf("by") + 3),
+                            DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm"));
+                    return new Deadline(deadlineDetails, false, by);
+                } catch (DateTimeParseException e) {
+                    throw new InvalidDateTimeException(ui);
+                }
+            } catch (StringIndexOutOfBoundsException e) {
+                throw new NoTaskDescriptionException(ui);
+            }
+
+
+        case EVENT:
+            try {
+                int atIndex = commandDetails.indexOf("/at") - 1 ;
+                if (atIndex <= 0) {
+                    throw new NoDateTimeException(ui);
+                }
+                String eventDetails = commandDetails.substring(0, atIndex);
+                if (eventDetails.isBlank()) {
+                    throw new NoTaskDescriptionException(ui);
+                }
+                try {
+                    LocalDateTime at = LocalDateTime.parse(commandDetails.substring(eventDetails.indexOf("at") + 3),
+                            DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm"));
+                    return new Event(eventDetails, false, at);
+                } catch (DateTimeParseException e) {
+                    throw new InvalidDateTimeException(ui);
+                }
+            } catch (StringIndexOutOfBoundsException e) {
+                throw new NoTaskDescriptionException(ui);
+            }
+
```

Comment:
The code in the `Deadline` and `Event` cases are very similar hence I think you could combine both into one case; alternatively, assuming all `taskType` inputs are valid, consider the `ToDo` case first and let the `Deadline` and `Event` be the default. To account for the 'by' and 'at' issue, maybe you can define an auxiliary delimiter string, if event then the string is 'at', else 'by'. This might help significantly reduce code duplication 👍🏼